modules/widgets/plasma-panel-colorizer.nix

```diff
@@ -0,0 +1,1097 @@
+                apply = getIndexFromEnum enumVals;
+              };
+            systemColorVariant =
+              let
+                enumVals = [
+                  "view"
+                  "window"
+                  "button"
+                  "selection"
+                  "tooltip"
+                  "complementary"
+                  "header"
+                ];
+              in
+              mkOption {
+                type = types.nullOr types.enum enumVals;
+                default = null;
+                example = "view";
+                description = ''
+                  System color variant for the widget outline.
+                '';
+                apply = getIndexFromEnum enumVals;
+              };
+            opacity = mkOption {
+              type = types.nullOr (types.numbers.between 0 1);
+              default = null;
+              example = 0.5;
+              description = "Opacity value";
+            };
+            width = mkOption {
+              type = types.nullOr types.ints.unsigned;
+              default = null;
+              example = 5;
+              description = "Width value";
+            };
+          };
+          shadow = {
+            color = mkOption {
+              type = types.nullOr types.str;
+              default = null;
+              example = "#7fb45c5c";
+              description = ''
+                Custom color for the widget shadow.
+              '';
+            };
+            size = mkOption {
+              type = types.nullOr types.ints.unsigned;
+              default = null;
+              example = 5;
+              description = "Size value";
+            };
+            horizontalOffset = mkOption {
+              type = types.nullOr types.int;
+              default = null;
+              example = 5;
+              description = "X offset value";
+            };
+            verticalOffset = mkOption {
+              type = types.nullOr types.int;
+              default = null;
+              example = (-1);
+              description = "Y offset value";
+            };
+          };
+        };
+      };
+      textAndIcons = {
+        enable = mkOption {
+          type = types.nullOr types.bool;
+          default = null;
+          example = true;
+          description = "Enable text and icons";
+        };
+        mode =
+          let enumVals = [ "static" "interval" ];
+          in mkOption {
+            type = types.nullOr types.enum enumVals;
+            default = null;
+            example = "static";
+            description = ''
+              Color mode for the widget text and icons.
+            '';
+            apply = getIndexFromEnum enumVals;
+          };
+        interval = mkOption {
+          type = types.nullOr types.ints.positive;
+          default = null;
+          example = 3000;
+          description = ''
+            Interval in milliseconds for the widget text and icons.
+          '';
+          colorMode =
```

Comment:
I believe this line causes the build to fail atm. There is a missing `};` (which does mean that there is one too many `};` somewhere else probably.

Reply:
I didn't notice that. Will fix it.

Reply:
I haven't really tested these options yet, but I will do it soon.